3Dmol/glmodel.js

```diff
@@ -1791,26 +1791,50 @@ $3Dmol.GLModel = (function() {
                         }
                     }else{//"or" and "and"
                         if(key == "and"){
-                            var and=sel[key];//array format
```

Comment:
So I think the goal here is to use selectedAtoms recursively which should handle within etc. statements unlike atomIsSelected.  This seems like a good idea.

Using index instead of serial should be more robust (index is suppose to be the position in the atoms array, whereas serial can be set in the PDB file).

Since we are keying on index, the implementation would be more efficient if they were all put in a Set and then the intersection/union taken.  This would also support refactoring or/and to eliminate some duplicate code (since the only difference is the set operation).  Using set lookup is O(1) vs the current O(n) search.

It would also be a good idea to memoize the result of the selectedAtoms call for even better efficiency.

Reply:
> Using index instead of serial should be more robust (index is suppose to be the position in the atoms array, whereas serial can be set in the PDB file).

Is `index` part of `AtomSpec` too? If so I can add it to #468.

I agree on the memoization & use of Set, I'll change this to use them.

Reply:
I'd rather leave it an internal property.